pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4363,11 +4363,20 @@ protected void internalHandleResult(AsyncResponse asyncResponse,
 
     protected void handleTopicPolicyException(String methodName, Throwable thr, AsyncResponse asyncResponse) {
         Throwable cause = thr.getCause();
-        if (!(cause instanceof WebApplicationException)
-                || !(((WebApplicationException) cause).getResponse().getStatus() == 307)) {
-            log.error("[{}] Failed to perform {} on topic {}",
-                    clientAppId(), methodName, topicName, cause);
+
+        if (cause instanceof WebApplicationException) {
+            int statusCode = ((WebApplicationException) cause).getResponse().getStatus();
+
+            // Print a warning level log when the topic does not exist, otherwise print an error level log.
```

Comment:
Nit: the code is very easy to understand, instead, bringing this description has a counterproductive effect, because TEMPORARY_REDIRECT will not print any logs.